File: prog8.py
def myatoi(s):
    if len(s)==0:
        return 0
    i=0
    while i<len(s) and s[i].isspace():
        i+=1

    result=0
    if s[i]=='-':
        sign=-1
        i+=1
    elif s[i]=='+':
        sign=1
        i+=1
    else:
        sign=1

    while i<len(s) and s[i].isdigit():
        digit=ord(s[i])-ord('0')
        result=10*result+digit
        i+=1
    return result*sign

x="   -123"
print (x,myatoi(x))
print (myatoi(x)*5)

File: prog8_.cpp
#include <iostream>
#include <string>
#include <cctype>
using namespace std;

int myatoi(string s)
{
    if (s.size()==0)
        return 0;

    int i=0;
    while (i<s.size() && isspace(s[i]))
    {
        i++;
    }

    int sign;
    int result =0;
    if (s[i] == '-')
    {
        sign=-1;
        i++;
    }
    else if (s[i] == '+')
    {
        sign=1;
        i++;
    }
    else
    {
        sign=1;
    }

    while (i<s.size() && isdigit(s[i]))
    {
        int digit = s[i]-'0';
        result = 10*result + digit;
        i++;
    }
    return result*sign;
}

int main()
{
    string x;
    cout << "enter string: ";
    cin >> x;
    cout << x << " : " << myatoi(x) << endl;
    return 0;
}
